src/Functions/serial.cpp

```diff
@@ -0,0 +1,171 @@
+#include <cmath>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <Columns/ColumnVector.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Functions/FunctionFactory.h>
+#include <Interpreters/Context.h>
+#include "Common/Logger.h"
+#include <Common/ZooKeeper/ZooKeeper.h>
+
+namespace DB {
+
+namespace ErrorCodes
+{
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+}
+
+class FunctionSerial : public IFunction
+{
+private:
+    mutable zkutil::ZooKeeperPtr zk{nullptr};
+    ContextPtr context;
+
+public:
+    static constexpr auto name = "serial";
+
+    explicit FunctionSerial(ContextPtr ctx) : context(ctx)
+    {
+        if (ctx->hasZooKeeper()) {
+            zk = ctx->getZooKeeper();
```

Comment:
But better to remove the check here, and the exception will be thrown automatically.

Reply:
If remove the check, an exception will be thrown when the client starts, isn't it?

Reply:
If you remove the check, an exception will be thrown when the client starts

"Cannot load data for command line suggestions: Code: 139. DB::Exception: Received from localhost:9000. DB::Exception: There is no Zookeeper configuration in server config. (NO_ELEMENTS_IN_CONFIG) (version 24.4.1.1)"

Reply:
Ok.